server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -21,188 +21,660 @@
+import org.elasticsearch.index.mapper.BitStringFieldMapper;
+import org.elasticsearch.index.mapper.DateFieldMapper;
+import org.elasticsearch.index.mapper.KeywordFieldMapper;
+import org.elasticsearch.index.mapper.TypeParsers;
+import org.jetbrains.annotations.Nullable;
 
+import io.crate.analyze.NumberOfReplicas;
+import io.crate.analyze.ParamTypeHints;
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.analyze.expressions.TableReferenceResolver;
+import io.crate.analyze.relations.FieldProvider;
+import io.crate.common.Booleans;
+import io.crate.common.collections.Maps;
 import io.crate.exceptions.RelationUnknown;
-import io.crate.exceptions.UnhandledServerException;
-import io.crate.metadata.IndexParts;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.GeneratedReference;
+import io.crate.metadata.GeoReference;
+import io.crate.metadata.IndexReference;
+import io.crate.metadata.IndexType;
 import io.crate.metadata.NodeContext;
 import io.crate.metadata.PartitionName;
+import io.crate.metadata.Reference;
+import io.crate.metadata.ReferenceIdent;
 import io.crate.metadata.RelationName;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.SimpleReference;
+import io.crate.metadata.table.Operation;
 import io.crate.replication.logical.metadata.PublicationsMetadata;
+import io.crate.server.xcontent.XContentHelper;
+import io.crate.sql.parser.SqlParser;
+import io.crate.sql.tree.CheckConstraint;
+import io.crate.sql.tree.ColumnPolicy;
+import io.crate.types.ArrayType;
+import io.crate.types.BitStringType;
+import io.crate.types.CharacterType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import io.crate.types.FloatVectorType;
+import io.crate.types.ObjectType;
+import io.crate.types.StorageSupport;
+import io.crate.types.StringType;
 
 @Singleton
 public class DocTableInfoFactory {
 
     private static final Logger LOGGER = LogManager.getLogger(DocTableInfoFactory.class);
 
     private final NodeContext nodeCtx;
+    private final ExpressionAnalyzer expressionAnalyzer;
+    private final CoordinatorTxnCtx systemTransactionContext;
 
     @Inject
     public DocTableInfoFactory(NodeContext nodeCtx) {
         this.nodeCtx = nodeCtx;
+        this.systemTransactionContext = CoordinatorTxnCtx.systemTransactionContext();
+        this.expressionAnalyzer = new ExpressionAnalyzer(
+            systemTransactionContext,
+            nodeCtx,
+            ParamTypeHints.EMPTY,
+            FieldProvider.UNSUPPORTED,
+            null
+        );
     }
 
     public DocTableInfo create(RelationName relation, ClusterState state) {
-        var metadata = state.metadata();
-        PublicationsMetadata publicationsMetadata = metadata.custom(PublicationsMetadata.TYPE);
+        return create(relation, state.metadata());
+    }
```

Comment:
Any reason to keep this instead of directly using the new `create(.. Metadata)` method below?

Reply:
Not really, I meant to inline it, but forgot